app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -85,6 +86,11 @@ class ScreenController {
     }
 
     fun showSettingsScreen(fragmentManager: FragmentManager) {
+        /**
+         *  Pause all the videos when transitioning to [SettingsFragment] to mitigate possible memory
+         *  leak when clearing data. See [WebViewCache.clear] as well as #1720
+         */
+        fragmentManager.webRenderFragment().webView?.pauseAllVideoPlaybacks()
```

Comment:
I'm uncomfortable that the `ScreenController` is calling directly into a `WebView`, but I'm trying to figure out a cleaner alternative and coming up dry.  @mcomella, @liuche do you have any thoughts here?

Reply:
Yeah I agree, ideally it would've been in the lifecycle method of it's respective fragment.

Reply:
Hm, can we add the webView handling code into `onStop` and then call that? I know it's further mixing our lifecycle vs manual fragment handling.

Reply:
@liuche You mean to override `EngineView.onStop()`?

Reply:
I wonder if `ScreenController` should become something more like `ScreenRepo`: what if the WebRenderFragment reacted to the current active screen and paused all the videos when the active screen is changed? Since we don't care about sync/async timing so much, I think that'd be fine.

fwiw, in #1797 (still needs to land), I made `currentActiveScreen` into a LiveData you can listen to.

Reply:
@mcomella I really like that solution.  @psymoon sorry for blocking this one so long, but what do you think about that suggestion?

Reply:
Sounds good to me